### ISSUE TITLE
Green lock (trusting the SSL certificate)

### ISSUE THREAD

**Issue by OliverJAsh** [typicode/hotel#94]
I am trying to get the green lock for the SSL certificate, which I have trusted.

I have the following error:

```
There are issues with the site's certificate chain (net::ERR_CERT_COMMON_NAME_INVALID).
```

Is there any way of providing valid certificates for use with local domains?

**Comment by typicode** [comment]
Hi @OliverJAsh,

Thank you for this issue, just to be sure I understand, you have trusted the self-signed certificate that comes with hotel and it doesn't work?

If that's it, what's the browser/OS?

> Is there any way of providing valid certificates for use with local domains?

AFAIK, I think you would need to generate a certificate for each local domain. And right now, hotel doesn't support custom certificates.

**Comment by OliverJAsh** [comment]
It does work, but for custom domains there is an error (see original post). I think I'm asking for support with custom certificates. Have you got any ideas on that?

**Comment by layerssss** [comment]
Exactly this should be very nice

https://github.com/typicode/hotel/pull/55#issuecomment-242954202

**Comment by OliverJAsh** [comment]
Hi @layerssss. Do you have any advice on how to use your [devhttps](https://github.com/layerssss/devhttps) tool along with hotel to achieve a similar effect here?

**Comment by layerssss** [comment]
I don't think you can use them together for now. Devhttps was designed for working with http://pow.cx/ which only exposes an HTTP server, while let other components handle DNS resolving. But hotel's working as an HTTP proxy, i can't make them work together easily.
